File: backend/app/core/roles.py
"""Roles de usuario."""

from enum import StrEnum


class Role(StrEnum):
    SUPER_ADMIN = "super_admin"  # Administrador global de la plataforma (sin tenant).
    ADMIN = "admin"  # Administrador de una empresa (tenant).
    EMPLOYEE = "employee"  # Usuario operativo de una empresa (tenant).


class TenantRole(StrEnum):
    """Roles que se pueden asignar a los usuarios de una empresa."""

    ADMIN = Role.ADMIN.value
    EMPLOYEE = Role.EMPLOYEE.value

File: frontend/src/lib/roles.ts
// Roles de usuario (mismos valores que backend/app/core/roles.py).

export type Role = 'super_admin' | 'admin' | 'employee'

/** Roles que se pueden asignar a los usuarios de una empresa. */
export type TenantRole = Exclude<Role, 'super_admin'>

export const TENANT_ROLES: TenantRole[] = ['admin', 'employee']

export const ROLE_LABELS: Record<Role, string> = {
  super_admin: 'Super Admin',
  admin: 'Administrador',
  employee: 'Empleado',
}
